packages/eslint-plugin/src/rules/require-array-sort-compare.ts

```diff
@@ -17,13 +31,35 @@ export default util.createRule({
     messages: {
       requireCompare: "Require 'compare' argument.",
     },
-    schema: [],
+    schema: [
+      {
+        type: 'object',
+        properties: {
+          ignoreStringArrays: {
+            type: 'boolean',
+          },
+        },
+      },
+    ],
   },
 
-  create(context) {
+  create(context, [options]) {
     const service = util.getParserServices(context);
     const checker = service.program.getTypeChecker();
 
+    /**
+     * Check if a given node is an array which all elements are string.
+     * @param node
+     */
+    function isStringArrayNode(node: TSESTree.ArrayExpression): boolean {
+      return node.elements.every(element => {
+        const type = checker.getTypeAtLocation(
+          service.esTreeNodeToTSNodeMap.get(element),
+        );
+        return util.getTypeName(checker, type) === 'string';
+      });
+    }
```

Comment:
this only works for array expressions that have `sort` directly called on them: `['a'].sort()`

A better solution would be to check the type of the `node` instead, and extract the type of the generic parameters.

All array types (`string[]` and `Array<string>`) are treated as `Array<string>` in code.
All tuple types (`[a, b, c]`) are treated as `Tuple<T1, T2, ...>` (a special variadic type).

```ts
const type = checker.getTypeAtLocation(
  service.esTreeNodeToTSNodeMap.get(node),
);
if (checker.isArrayType(type) || checker.isTupleType(type)) {
  const typeArgs = checker.getTypeArguments(type);
  return checker.getTypeArguments(type).every(
    arg => util.getTypeName(checker, arg) === 'string',
  );
}
return false;
```